internal/xds/bootstrap/bootstrap.go

```diff
@@ -87,50 +84,125 @@ func (cc ChannelCreds) String() string {
+	// XDSServers contains the list of server configurations for this authority.
+	XDSServers []*ServerConfig `json:"xds_servers,omitempty"`
+}
+
+// Equal returns true if a equals other.
+func (a *Authority) Equal(other *Authority) bool {
+	switch {
+	case a == nil && other == nil:
+		return true
+	case (a != nil) != (other != nil):
+		return false
+	case a.ClientListenerResourceNameTemplate != other.ClientListenerResourceNameTemplate:
+		return false
+	case !slices.EqualFunc(a.XDSServers, other.XDSServers, func(a, b *ServerConfig) bool { return a.Equal(b) }):
+		return false
+	}
+	return true
+}
+
+// ServerConfig contains the configuration to connect to a server.
+type ServerConfig struct {
+	serverURI      string
+	channelCreds   []ChannelCreds
+	serverFeatures []string
 
 	// As part of unmarshalling the JSON config into this struct, we ensure that
 	// the credentials config is valid by building an instance of the specified
-	// credentials and store it here as a grpc.DialOption for easy access when
-	// dialing this xDS server.
+	// credentials and store it here for easy access.
+	selectedCreds   ChannelCreds
 	credsDialOption grpc.DialOption
 
-	// IgnoreResourceDeletion controls the behavior of the xDS client when the
-	// server deletes a previously sent Listener or Cluster resource. If set, the
-	// xDS client will not invoke the watchers' OnResourceDoesNotExist() method
-	// when a resource is deleted, nor will it remove the existing resource value
-	// from its cache.
-	IgnoreResourceDeletion bool
+	cleanups []func()
+}
+
+// ServerURI returns the URI of the management server to connect to.
+func (sc *ServerConfig) ServerURI() string {
+	return sc.serverURI
+}
+
+// ChannelCreds returns the credentials configuration to use when communicating
+// with this server. Also used to dedup servers with the same server URI.
+func (sc *ServerConfig) ChannelCreds() []ChannelCreds {
+	return sc.channelCreds
+}
+
+// ServerFeatures returns the list of features supported by this server. Also
+// used to dedup servers with the same server URI and channel creds.
+func (sc *ServerConfig) ServerFeatures() []string {
+	return sc.serverFeatures
+}
 
-	// Cleanups are called when the xDS client for this server is closed. Allows
-	// cleaning up resources created specifically for this ServerConfig.
-	Cleanups []func()
+// IgnoreResourceDeletion returns true if this server supports a feature where
+// the xDS client can ignore resource deletions from this server, as described
+// in gRFC A53.
+//
+// This feature controls the behavior of the xDS client when the server deletes
+// a previously sent Listener or Cluster resource. If set, the xDS client will
+// not invoke the watchers' OnResourceDoesNotExist() method when a resource is
+// deleted, nor will it remove the existing resource value from its cache.
+func (sc *ServerConfig) IgnoreResourceDeletion() bool {
```

Comment:
optional:
```suggestion
func (sc *ServerConfig) ServerFeaturesIgnoreResourceDeletion() bool {
```
? To make the callsites look nicer?